build-scripts/package-tests.cake

```diff
@@ -0,0 +1,382 @@
+            "netcoreapp2.1/mock-assembly.dll",
+            new ExpectedResult("Failed")
+            {
+                Total = 37,
+                Passed = 23,
+                Failed = 5,
+                Warnings = 0,
+                Inconclusive = 1,
+                Skipped = 7
+            }));
+
+        PackageTests.Add(new PackageTest(1, "Run mock-assembly targeting .NET Core 3.1",
+            "netcoreapp3.1/mock-assembly.dll",
+            new ExpectedResult("Failed")
+            {
+                Total = 37,
+                Passed = 23,
+                Failed = 5,
+                Warnings = 0,
+                Inconclusive = 1,
+                Skipped = 7
+            }));
+
+        PackageTests.Add(new PackageTest(1, "Run both copies of mock-assembly together",
+            "netcoreapp2.1/mock-assembly.dll netcoreapp3.1/mock-assembly.dll",
+            new ExpectedResult("Failed")
+            {
+                Total = 2 * 37,
+                Passed = 2 * 23,
+                Failed = 2 * 5,
+                Warnings = 0,
+                Inconclusive = 2 * 1,
+                Skipped = 2 * 7
+            }));
+
+        // TODO: Should we have a .NET 5.0 test?
+        //PackageTests.Add(new PackageTest(1, "Run mock-assembly.dll under .NET 5.0",
+        //    "net5.0/mock-assembly.dll",
+        //    new ExpectedResult("Failed")
+        //    {
+        //        Total = 32,
+        //        Passed = 19,
+        //        Failed = 5,
+        //        Warnings = 0,
+        //        Inconclusive = 1,
+        //        Skipped = 7
+        //    }));
+    }
+}
+
+public class ZipPackageTester : NetFXPackageTester
+{
+    public ZipPackageTester(BuildParameters parameters) : base(parameters) { }
+
+    protected override string PackageName => _parameters.ZipPackageName;
+    protected override FilePath PackageUnderTest => _parameters.ZipPackage;
+    protected override string PackageTestDirectory => _parameters.ZipTestDirectory;
+    protected override string PackageTestBinDirectory => PackageTestDirectory + "bin/net20/";
+    protected override string ExtensionInstallDirectory => PackageTestBinDirectory + "addins/";
+
+    protected override void InstallEngineExtension(string extension)
+    {
+        _context.NuGetInstall(extension,
+            new NuGetInstallSettings() { OutputDirectory = ExtensionInstallDirectory });
+    }
+}
+
+public class NuGetNetFXPackageTester : NetFXPackageTester
+{
+    public NuGetNetFXPackageTester(BuildParameters parameters) : base(parameters) { }
+
+    protected override string PackageName => _parameters.NuGetNetFXPackageName;
+    protected override FilePath PackageUnderTest => _parameters.NuGetNetFXPackage;
+    protected override string PackageTestDirectory => _parameters.NuGetNetFXTestDirectory;
+    protected override string PackageTestBinDirectory => PackageTestDirectory + "tools/";
+    protected override string ExtensionInstallDirectory => _parameters.TestDirectory;
```

Comment:
Should this be TestDirectory? Or TempDirectory, where the package is being installed? Maybe I'm missing something around how this works. 😕